mucommander-core/src/main/java/com/mucommander/ui/main/MainFrame.java

```diff
@@ -205,8 +206,16 @@ public Insets getInsets() {
 
         executor.execute(() -> {
             // Create menu bar (has to be created after toolbar) - ok, but why?
+            // PSko - I guess it is related to loading Actions and that icons
+            // for toolbar action should have icons, but in menu they should not.
+            // However, still I don't get how setting the icon to null here in MenuToolkit#addMenuItem
+            // impacts menu icons....... if nullify is commented-out there, then icons all of sudden
```

Comment:
> However, still I don't get how setting the icon to null here in MenuToolkit#addMenuItem impacts menu icons.......

This part, that setting the icon to null impacts menu icons, makes sense to me but yeah, from a quick look it's not clear to me either how come that when something within the toolbar thread might affect the menu item - maybe it (the toolbar thread) changes the action and the menu item listen to changes on the actions and then sets its icon accordingly..